### ISSUE TITLE
🐛 [BUG]: Nuxt3 Hydration attribute mismatch

### ISSUE THREAD

**Issue by iyume** [bcakmakoglu/vue-flow#1350]
### Is there an existing issue for this?

- [X] I have searched the existing issues and this is a new bug.

### Current Behavior

https://vueflow.dev/examples/dnd.html

Setup a new project on Drag & Drop example caused warning:

![screenshot-20240314-163505](https://github.com/bcakmakoglu/vue-flow/assets/48091591/acaae1f7-25d3-4c1e-bedc-87e5fd330788)

### Expected Behavior

No warning

### Steps To Reproduce

setup new project

```json
{
  "name": "nuxt-app",
  "private": true,
  "type": "module",
  "scripts": {
    "build": "nuxt build",
    "dev": "nuxt dev",
    "generate": "nuxt generate",
    "preview": "nuxt preview",
    "postinstall": "nuxt prepare"
  },
  "dependencies": {
    "@vue-flow/background": "^1.3.0",
    "@vue-flow/core": "^1.33.4",
    "class-variance-authority": "^0.7.0",
    "clsx": "^2.1.0",
    "lucide-vue-next": "^0.357.0",
    "nuxt": "^3.10.3",
    "radix-vue": "^1.5.2",
    "tailwind-merge": "^2.2.1",
    "tailwindcss-animate": "^1.0.7",
    "vue": "^3.4.21",
    "vue-router": "^4.3.0"
  },
  "devDependencies": {
    "@nuxtjs/tailwindcss": "^6.11.4",
    "shadcn-nuxt": "^0.10.0",
    "typescript": "^5.4.2"
  }
}

```

### Relevant log output

```shell
// Nuxt

Debugger attached.
Nuxt 3.10.3 with Nitro 2.9.3

  ➜ Local:    http://localhost:3000/
  ➜ Network:  use --host to expose

Debugger attached.
[nuxt:tailwindcss] ℹ Using Tailwind CSS from ~/assets/css/tailwind.css
  ➜ DevTools: press Shift + Alt + D in the browser (v1.0.8)

[nuxt:tailwindcss] ℹ Tailwind Viewer: http://localhost:3000/_tailwind/
ℹ Vite client warmed up in 1330ms
ℹ Vite server warmed up in 1491ms
[nitro] ✔ Nuxt Nitro server built in 590 ms


// Chrome

[vite] connected.
client.ts:173
<Suspense> is an experimental feature and its API will likely change.
runtime-core.esm-bundler.js:1444
[Vue warn]: Hydration attribute mismatch on pattern#pattern-vue-flow-4 
  - rendered on server: id="pattern-vue-flow-4"
  - expected on client: id="pattern-vue-flow-0"
  Note: this mismatch is check-only. The DOM will not be rectified in production due to performance overhead.
  You should fix the source of the mismatch. 
  at <Background size=2 gap=20 pattern-color="#BDBDBD" > 
  at <DropzoneBackground style= {backgroundColor: 'transparent', transition: 'background-color 0.2s ease'} > 
  at <VueFlow nodes= (0) [] onDragover=fn<onDragOver> onDragleave=fn<onDragLeave> > 
  at <App key=3 > 
  at <NuxtRoot>
runtime-core.esm-bundler.js:47
```

**Comment by bcakmakoglu** [comment]
Can you give me some info on your nuxt config?

**Comment by iyume** [comment]
```
// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
  devtools: { enabled: true },
  modules: ["@nuxtjs/tailwindcss", "shadcn-nuxt"],
  shadcn: {
    /**
     * Prefix for all the imported component
     */
    prefix: "",
    /**
     * Directory that the component lives in.
     * @default "./components/ui"
     */
    componentDir: "./components/ui",
  },
});
```

**Comment by bcakmakoglu** [comment]
Thanks, I'll investigate this issue :)

**Comment by bcakmakoglu** [comment]
I can't reproduce this issue with the given steps, no hydration errors appear for me.

Could you provide a reproduction in the form of a repo or a sandbox?

**Comment by iyume** [comment]
Setup:

```
mkdir test
cd test
npx nuxi@latest init . --force
pnpm add @vue-flow/core
pnpm add @vue-flow/background
```

File structure:

```
.
├── README.md
├── app.vue
├── assets
│   └── main.css
├── components
│   ├── DropzoneBackground.vue
│   └── Sidebar.vue
├── composables
│   └── useDnD.js
├── nuxt.config.ts
├── package.json
├── pnpm-lock.yaml
├── public
│   └── favicon.ico
├── server
│   └── tsconfig.json
└── tsconfig.json
```

`useDnD.js` line 26:

```
export function useDragAndDrop() {
```

Remove all relative imports and use nuxt auto imports.

Add `import "~/assets/main.css";` in `app.vue`

Change `main.css` .dndflow with `height: 100vh;`

Start with:

```
npx nuxi dev
```

**Comment by iyume** [comment]
A clear package.json

```json
{
  "name": "nuxt-app",
  "private": true,
  "type": "module",
  "scripts": {
    "build": "nuxt build",
    "dev": "nuxt dev",
    "generate": "nuxt generate",
    "preview": "nuxt preview",
    "postinstall": "nuxt prepare"
  },
  "dependencies": {
    "@vue-flow/background": "^1.3.0",
    "@vue-flow/core": "^1.33.4",
    "nuxt": "^3.10.3",
    "vue": "^3.4.21",
    "vue-router": "^4.3.0"
  }
}
```

**Comment by iyume** [comment]
A minimal code to reproduce:

```vue
<script setup>
import { VueFlow } from "@vue-flow/core";
</script>

<template>
  <div style="height: 100vh; width: 100vw">
    <VueFlow> </VueFlow>
  </div>
</template>

<style>
/* import the necessary styles for Vue Flow to work */
@import "@vue-flow/core/dist/style.css";

/* import the default theme, this is optional but generally recommended */
@import "@vue-flow/core/dist/theme-default.css";
</style>
```

![图片](https://github.com/bcakmakoglu/vue-flow/assets/48091591/7f8ba488-97c7-4f83-a635-1e86b38d7055)

**Comment by bcakmakoglu** [comment]
Well... another symptom of bad API decisions I made early on 😆

This should be patched and resolved with the next release but a proper solution for this will be introduced in the next major update.

**Comment by bcakmakoglu** [comment]
Fixed with `1.33.5`

**Comment by Cybrarist** [comment]
apparently this issue exists in nuxt v4 too

to fix it, either disable ssr in nuxt config file  or wrap vueflow with <clientonly> component
